apis/v1alpha2/httproute_types.go

```diff
@@ -598,6 +614,24 @@ type HTTPRequestHeaderFilter struct {
 	Remove []string `json:"remove,omitempty"`
 }
 
+// HTTPRequestRedirect defines configuration for the RequestRedirect filter.
+type HTTPRequestRedirect struct {
+	// AutoHTTPS if set to true redirects any incoming HTTP requests to HTTPS.
```

Comment:
> Gateway owner wants to perform HTTP->HTTPS on all requests. Could this be a property of GatewayClass?

Would it be possible to define a route for `*` and set auto_https?

Reply:
Even better if our use case also supports ACME where we probably want to redirect eveything *but* /.wellknown/acme

Reply:
> Even better if our use case also supports ACME where we probably want to redirect eveything but /.wellknown/acme

Solid point! 

> Would it be possible to define a route for * and set auto_https?

Gateway admin can but it has limitations. The most specific route matches a request and if that doesn't have the filter defined, then HTTP requests will be accepted.

Reply:
But in that case you could have a listener on 80 that only selected the auto redirect route and a listener on 443 that selected all the other routes? This seems like it could become a common pattern.